src/Schema/Events/Search.ts

```diff
@@ -0,0 +1,89 @@
+import { ActionType } from "."
+import { ContextModule } from "../Values/ContextModule"
+import { OwnerType, PageOwnerType } from "../Values/OwnerType"
+
+/**
+ * Schemas describing Search events
+ * @packageDocumentation
+ */
+
+/**
+ * A user focuses on a search box
+ *
+ * This schema describes events sent to Segment from [[FocusedOnSearchInput]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "focusedOnSearchInput",
+ *    context_module: "header",
+ *    context_page_owner_type: "home"
+ *  }
+ * ```
+ */
+export interface FocusedOnSearchInput {
+  action: ActionType.focusedOnSearchInput
+  context_module: ContextModule
+  context_page_owner_type: PageOwnerType
+  context_page_owner_id?: string
+  context_page_owner_slug?: string
+}
+
+/**
+ * A user selects an item from search
+ *
+ * This schema describes events sent to Segment from [[SelectedItemFromSearch]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "selectedItemFromSearch",
+ *    context_module: "priceEstimate",
+ *    context_page_owner_type: "consign",
+ *    owner_type: "artist",
+ *    owner_id: "5df3e3fa485efe0012c37055",
+ *    owner_slug: "andy-warhol",
+ *    query: "andy warhol"
+ *  }
+ * ```
+ */
+export interface SelectedItemFromSearch {
+  action: ActionType.selectedItemFromSearch
+  context_module: ContextModule
+  context_page_owner_type: PageOwnerType
+  context_page_owner_id?: string
+  context_page_owner_slug?: string
+  destination_page_owner_type?: PageOwnerType
+  destination_page_owner_id?: string
+  destination_page_owner_slug?: string
+  owner_type: OwnerType
+  owner_id: string
+  owner_slug: string
+  query: string
+}
+
+/**
+ * A user searches with no results
+ *
+ * This schema describes events sent to Segment from [[SearchedWithNoResults]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "searchedWithNoResults",
+ *    context_module: "header",
+ *    context_page_owner_type: "home",
+ *    destination_page_owner_type: "search",
+ *    query: "andhuwe"
+ *  }
+ * ```
+ */
+export interface SearchedWithNoResults {
+  action: ActionType.searchedWithNoResults
+  context_module: ContextModule
+  context_page_owner_type: PageOwnerType
+  context_page_owner_id?: string
+  context_page_owner_slug?: string
+  destination_page_owner_type?: PageOwnerType
```

Comment:
Is it possible for there to be a destination when there are no results? If there is one `destination_page_owner_type` should probably be required.

Reply:
This is also because of the consign page - if there are no results, you won't go to a separate page.